### PR TITLE
Cleanup tasks

### PR DESCRIPTION
**Pull request by jakemac53**
- Move to `modifyFile` from `updateFile` analyzer api.
- Remove an unnecessary dependency_override in build_runner.

### CODE REVIEW

**Comment by jakemac53** [comment]
Note that I reverted the analyzer change after discussion with the analyzer folks - it looks like there was a mistake and `updateFile` is the one we should continue to use, and `modifyFile` will be deprecated.

**Review by natebosch — APPROVED** [review]

**Review by natebosch — APPROVED** [review]

File: build_resolvers/lib/src/build_asset_uri_resolver.dart [hunk 1]
```diff
@@ -125,7 +125,7 @@ class BuildAssetUriResolver extends UriResolver {
         return _AssetState(path, _cachedAssetDependencies[id]!);
       }
       if (isChange) {
-        resourceProvider.updateFile(path, content);
+        resourceProvider.modifyFile(path, content);
```
Comment (kevmoo) [inline build_resolvers/lib/src/build_asset_uri_resolver.dart:128]: sure the pkg:analyzer version we have has this?

Comment (jakemac53) [reply build_resolvers/lib/src/build_asset_uri_resolver.dart:128]: yes

**Comment by jakemac53** [comment]
cc @scheglov it looks like the new API doesn't throw the `InconsistentAnalysisException` as expected, is that intentional? See https://github.com/dart-lang/build/pull/3199/checks?check_run_id=3792799647#step:6:98.

**Comment by scheglov** [comment]
Yes, it is intentional. We now try to keep libraries that are not affected by the change to a file. So, we never discard the whole session. But we mark individual libraries are invalid.

See https://dart-review.googlesource.com/c/sdk/+/214502

However it seems that I did not change places where we used to check for session to checking this new flag. I will need to make these updates.

**Comment by jakemac53** [comment]
Ok so I think this test would need some additional fixes then - it will need to modify the file and re-invalidate it.

I tried adding this line to replace where we used to do a `isLibrary` check on another library (that used to invalidate the session):

```dart
        (x.library!.session.resourceProvider as MemoryResourceProvider)
            .modifyFile('/a/web/main.dart', 'int y;');
```

That doesn't appear to work today but it sounds like it might be related to the other issue you mentioned?

**Comment by jakemac53** [comment]
Just skipped those tests for now

**Comment by scheglov** [comment]
Yes, methods that accept `LibraryElement` should be updated to check that the element is still valid, and return an `InvalidResult`, in my work-in-progress CL it would be:
```dart
/// The type of [InvalidResult] returned when the given element was invalidated
/// since it was created and returned to the client, because of a change to
/// a file that affected it.
///
/// Clients may not extend, implement or mix-in this class.
class InvalidatedElementResult
    implements
        InvalidResult,
        SomeParsedLibraryResult,
        SomeResolvedLibraryResult {}
```

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/215482

**Comment by natebosch** [comment]
@scheglov - is the analyzer team still considering a change in this design?

**Comment by scheglov** [comment]
No, we don't.
https://dart-review.googlesource.com/c/sdk/+/230040 was implemented instead.

**Comment by scheglov** [comment]
Both should throw, as you can see from https://dart-review.googlesource.com/c/sdk/+/230040/3/pkg/analyzer/test/src/dart/analysis/session_test.dart#300

But be aware that `modifyFile` only changes the file system, and `AnalysisDriver` does not watch the file system, it is up to the higher level clients to do and call `AnalysisDriver.changeFile()`.

**Pull request merged**